### DISCUSSION TITLE
KIN buy-in module parameter in KIN KRE 3.1

### DISCUSSION THREAD

**Discussion by Polype01** [kinfoundation/kin-rfcs#138] [_ Archive - KRE 3.1 Discussion]
Hello,

I really like the adjustment of the KIN KRE 3.1. 

I am still missing a KIN buy-in track in the KRE algorithm

The KIN KRE 3.1 proposal looks like this
An app gets more rewards if it has higher spends than other apps: Spendapp / Spendmax
An app gets more rewards if its users hold individual higher balances: Balanceapp / Balancemax
An app gets more rewards if its AU is higher than other Apps: AUapp / AUmax

how about
An app gets more rewards if it has higher spends than other apps: Spendapp / Spendmax
An app gets more rewards if its users hold individual higher balances: Balanceapp / Balancemax
An app gets more rewards if its AU is higher than other Apps: AUapp / AUmax
**An app gets more rewards if its KIN buy in-app amount is higher than other Apps: Kinbuy / Kinbuymax**

Apps that allow users to buy KIN (integration with Simplex/Ramp) should be also rewarded as they contribute in increasing the Real demand for KIN currency. By allowing users to buy KIN, the apps contribute positively in the KIN economy (and increasing KIN's value overall)

Of course, ithe prerequisite is that it should be easy to track accurately the correct amounts of KIN bought from those third parties (Ramp, Simplex..)

Remark: I am missing the weight of the three parameters of the new KIN KRE algorithm. Is it strictly
33,33% * (Spendapp / Spendmax)
33,33% * (Balanceapp / Balancemax)
33,33% * (AUapp / AUmax)
?

Kind regards,
Polype01

**Comment by alexkruglov** [comment]
I think this is an interesting idea, but not sure today's environment is conducive to making it practical. in other words, ramp is the only solution available and is facing implementation challenges (geography and fees). I would rather see a goal that the proposal hints at - e.g, usage of more of Solana SDK such as the Phantom/Solflare wallet. Wouldn't make sense to punish apps whose users use the Phantom wallet and buy Kin via DeFi

**Reply by Polype01** [reply]
One of KIN's issues currently is the price. 

KIN needs to grow in value organically (increase of real demand) and the implementation of "KIN buy-in modules" such as Simplex or Ramp should be pushed more actively to help KIN appreciate. Anything that can help KIN growing in value should be a priority imo.

If the buying of KIN is not officially integrated in the KIN KRE algorithm, then it won't be implemented de facto by the devs.
Maybe the weight of the KIN buy-in modue should be less than the three other criteria.

How about
30% * (Spendapp / Spendmax)
30% * (Balanceapp / Balancemax)
30% * (AUapp / AUmax)
10% (Kinbuy / Kinbuymax)

This would be a nice incentive to still implement the buying module but not a catastrophy if the apps choose not to implement it

Of course, as always, in order to give the users the strong will to buy KIN, we need more meaningful spends in apps first.

**Comment by Freedom-kin** [comment]
Hi Polype,

My question is,,,,what merit is it for an app to implement a buy module?
It's a tool and makes it easy for users to buy Kin in app.

It's the same as a regular store that has a payment terminal. Customers can pay more easily.

I don't see the reason why apps should be rewarded extra for having this.
It's in the apps own interest to implement it.

Also, it adds an extra factor to the already high number of factors. It's getting more and more complicated.

**Reply by Polype01** [reply]
The more KIN is bought, the more positive impact it will have on its value overall.
If an app allows a user to buy 100$ of KIN via Ramp or Simplex, it should be rewarded for having increased the overall value of KIN (even by 100$)
It's the apps own interest to implement it but not all current apps did implement it (yet?)
Complicated? why?

By proposing KIN to be earned if users buy KIN, the apps will be incentivized to 
1) implement the module
2) propose meaningful spends in order to give the user the will to buy KIN to spend it in app

**Reply by Freedom-kin** [reply]
What I'm saying is, a buy module by itself won't buy more Kin. It is the SPENDS **opportunities** that tempt the user to purchase more Kin. The buy module just makes it easier. So it is the SPENDS that we have to reward, then they will automatically implement the buy module in their app. An extra reward is then not necessary.

The problem is solved if you reward the net SPENDS with 30% KRE (with a cap). I can't think of a simpler KRE. When developers see their SPENDS being rewarded they will do anything to increase it. The implementation of a buy module is then a logical choice.

It could also be interesting for larger apps.
**The idea that you get a 30% bonus on top of your net turnover must be attractive.**

**Comment by alexkruglov** [comment]
The challenge with having these incentives is unintended consequences. We want users to buy Kin, but we also want Kin to flow freely along the ecosystem. It is after all a cryptocurrency. The unintended consequences here would be apps competing for incentivizing users to buy the Kin and hence preventing users from simply bringing in Kin that was purchased elsewhere (which is good for the ecosystem)
